### DISCUSSION TITLE
Running local fork during app development?

### DISCUSSION THREAD

**Discussion by Brodzko** [TanStack/react-charts#231] [Q&A] [answered]
Hi!

I forked the repo in an attempt to debug a chart I want to add to my app but I can't seem to get everything running smoothly. Reproduction steps:

- fork this repo
- checkout a new branch based on `beta`
- run `yarn install`
- run `npm link`in this lib's root folder (I'm using `npm` in my app)
- run `npm link react-charts` in my app root folder
- run `yarn watch` in this lib's root folder
- run my app

This creates the symlink correctly and my app compiles but when I try to render the chart I get a `react-hooks error` (see [this discussion](https://github.com/facebook/react/issues/13991). Based on this I guess `react-charts` build process includes React in its bundle which in turn causes my app to have two of them, or have I missed something else?

Would moving `react` and `react-dom` to `peerDependencies` solve this?

Bonus suggestions: Why not build the project into a single `build` folder but rather generate everything in root? Not only does this clutter the root folder but symlinking would also be more straightforward?

Thanks for answering!

**Comment by tannerlinsley** [comment, marked as answer]
This has more to do with yarn/npm/webpack's difficulty with symlinking resulting in multiple react copies. It's difficult to get around regardless of how you set things up. You'll notice that the simple example uses rescripts and has a few manual build workarounds to get rid of this issue. It's annoying to say the least. You could start by looking there.
